build.sbt

```diff
@@ -165,9 +165,11 @@ lazy val packagingSettings = Seq(
     ("./version docker" !!).trim
   },
   (defaultLinuxInstallLocation in Docker) := "/marathon",
+  maintainer := "Mesosphere Package Builder <support@mesosphere.io>",
   dockerCommands := {
-    // kind of a work-around; we want our mesos install and jdk install to come earlier so that Docker can cache them
-    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(2)
+    // kind of a work-around; we need our chown /marathon command to come after the WORKDIR command, and installation
+    // commands to preceed adding the Marthon artifact so that Docker can cache them
+    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(3)
```

Comment:
I don't understand why splitAt(2) changed to splitAt(3) :/

Reply:
>  we need our chown /marathon command to come after the WORKDIR command

:)

Reply:
I thought about switching it to looking for the WORKDIR command; that would be clearer, but it was more changes.

Reply:
Is this better for you?

```scala
    val (prefixCommands, restCommands) = dockerCommands.value.splitAt(dockerCommands.value.indexWhere(_.makeCommand.startsWith("WORKDIR ")) + 1)
```

Reply:
That's absolutely perfect 😍